Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -47,10 +61,11 @@ class OpenSearchEngine {
 
     private func getURLFromTemplate(searchTemplate: String, query: String) -> NSURL? {
         let allowedCharacters = NSCharacterSet(charactersInString: SearchTermsAllowedCharacters)
-
         if let escapedQuery = query.stringByAddingPercentEncodingWithAllowedCharacters(allowedCharacters) {
             let urlString = searchTemplate.stringByReplacingOccurrencesOfString("{searchTerms}", withString: escapedQuery, options: NSStringCompareOptions.LiteralSearch, range: nil)
-            return NSURL(string: urlString)
+            if let encodedUrlString = urlString.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.symbollessURLAllowedSet()) {
```

Comment:
That's saying we want to allow any character that isn't a symbol. This will filter out and symbol characters or UTF16/32 characters. Having just this will filter out the & and ? symbols in the URL which is why I also union this with what's acceptable in a URL.